tests/Microsoft.Identity.Test.Common/TestConstants.cs

```diff
@@ -460,4 +460,34 @@ internal static class Adfs2019LabConstants
         public const string ADFS2019ClientSecretURL = "https://buildautomation.vault.azure.net/secrets/ADFS2019ClientCredSecret/";
         public const string ADFS2019ClientSecretName = "ADFS2019ClientCredSecret";
     }
+
+    public enum MsiAzureResource
+    {
+        WebApp,
+        Function,
+        Vm,
+        AzureArc,
+        CloudShell,
+        ServiceFabric
+    }
+
+    public enum SendMSIHeader
+    {
+        Original,
+        WithWrongValue,

```

Comment:
We have not really discussed negative testing - but in negative scenarios we are: 

1. testing to ensure correct, easy to understand exceptions are thrown back to user (this is unit test territory, try not to use integration tests for these since they are expensive)
2. ensure that the integration point (e.g. MSI service) behaves in a consistent way when hit with "bad" values. 

I'd like to better understand what the negative test strategy is first before reviewing code. Also, to keep PRs reasonable, aim to deliver the smallest piece of code that provides value - the positive tests are enough. Negative tests should be added as separate PR.

Reply:
removed negative cases